ols/src/llms/llm_loader.py

```diff
@@ -58,28 +59,47 @@ def __init__(
         self,
         provider: Optional[str] = None,
         model: Optional[str] = None,
-        url: Optional[str] = None,
         params: Optional[dict] = None,
         logger=None,
     ) -> None:
         """Initialize loader using provided provider, model, and other parameters."""
         self.logger = logger if logger is not None else Logger("llm_loader").logger
-        if provider is None:
+        if not provider:
```

Comment:
i don't know what python style guides say, but i find `provider is None` to be more readable/obvious than `not provider`, given that provider isn't a boolean.

Reply:
`if not provider` check for any "falsy" value, but I agree `provider is None` is explicit here, reverting